test/api/fake_api.dart

```diff
@@ -29,11 +29,11 @@ class _PreparedSuccess extends _PreparedResponse {
 /// An [http.Client] that accepts and replays canned responses, for testing.
 class FakeHttpClient extends http.BaseClient {
 
-  http.BaseRequest? lastRequest;
+  List<http.BaseRequest> requestHistory = [];
```

Comment:
It might be helpful to make this a getter that returns an `Iterable`, which is backed by a private field that's a List? Then consumers of the request history can't mess with things by modifying the list.